### PR TITLE
UX: Multichain: Use Modal for AccountListMenu

### LINKED ISSUE #19834: UX: Multichain: Convert Account, Network, and Details Popovers to Modals
### Describe the bug

The Account, Network, and Account Details UI were written with the legacy `Popover` component since the new `Modal` wasn't ready.  With the new `Modal` ready, we should convert those components to a `Modal`.

- [ ] Account Menu
- [ ] Network Menu
- [ ] Account Details

### Steps to reproduce

N/A

### Error messages or log output

_No response_

### Version

develop

### Build type

None

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by darkwing**
## Explanation

Turns the AccountListMenu from a Popover into a Modal.

Fixes: #19834

## ToDo

- [ ]  Add a dozen accounts and ensure the modal still looks good in popup (doesn't require the user to scroll through the entire popup).
- [ ] Get E2Es passing.  I believe the only reason they're failing now is checks for classes that are `.popover-*`, so we'll need to update those CSS selectors

## Screenshots/Screencaps

SCR-20230628-kzqs

## Manual Testing Steps

Open the AccountListMenu, ensure it works just as it did before.

## Pre-merge author checklist

- [ ] I've clearly explained:
  - [ ] What problem this PR is solving
  - [ ] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by SaraCheikh** [comment]
this looks good!

**Review by SaraCheikh — DISMISSED** [review]

File: ui/components/multichain/account-list-menu/account-list-menu.js [hunk 1]
```diff
@@ -83,201 +78,220 @@ export const AccountListMenu = ({ onClose }) => {
...
-          ) : null}
-          {/* Account list block */}
-          <Box className="multichain-account-menu-popover__list">
-            {searchResults.length === 0 && searchQuery !== '' ? (
-              <Text
+    <Modal isOpen onClose={onClose}>
+      <ModalOverlay />
+      <ModalContent
+        className="multichain-account-menu-popover"
+        modalDialogProps={{ padding: 0 }}
```
Comment (georgewrmarshall) [inline ui/components/multichain/account-list-menu/account-list-menu.js:85]: I need to adjust this in the component but this should remove the extra scroll bar

```suggestion
        modalDialogProps={{ padding: 0, marginBottom: 0 }}
```

Screenshot 2023-06-28 at 11 16 03 AM

Screenshot 2023-06-28 at 11 14 22 AM

**Review by georgewrmarshall — COMMENTED** [review]
One small update otherwise LGTM

**Review by NidhiKJha — COMMENTED** [review]
Looks good! Just need to fix the popover inside the accounts menu list for account details and view on etherscan link.

https://github.com/MetaMask/metamask-extension/assets/39872794/e9f742f7-b0d7-4d8f-83ee-0e736281b161

**Review by alokeutpal — APPROVED** [review]

**Review by albertolive — APPROVED** [review]

**Review by montelaidev — APPROVED** [review]
Looks good

**Pull request merged**